generators/app/templates/src/.dev/login/loginLogic.js

```diff
@@ -218,26 +262,58 @@ class GeotabLogin {
                     window.speechSynthesis.speak(utterThis);
                 }
             },
-            notify: function (message, title, id, jsonData, permanent) {
-                var notification,
-                    options = {
-                        tag: id,
-                        body: message,
-                        data: jsonData
-                    };
-
-                if (!('Notification' in window)) {
-                    console.log('This browser does not support notifications');
-                } else if (Notification.permission === 'granted') {
-                    notification = new Notification(title, options);
-                } else if (Notification.permission !== 'denied') {
-                    Notification.requestPermission(function (permission) {
-                        if (permission === 'granted') {
+            notification: function (){                
+                return{
+                    hasPermission: function(){
+                        var permission = false;
+                        if(Notification.permission === 'granted')
+                        {
+                            permission == true;
+                        }
+                        return permission;
+                    },
+                    requestPermission: function(){  
+                        var r = Notification.requestPermission().then(function(result){                            
```

Comment:
I think we need to return the promise and result here.
```suggestion
                        return Notification.requestPermission();                            
```